File: prime-number/cmd.go
package main

import (
    "fmt"
    "os"
    "strconv"
)

func main() {
    primeNumberCount := 100;
    if len(os.Args) > 1 {
        i, err := strconv.Atoi(os.Args[1])
        if err != nil {
            panic(err)
        } else {
            primeNumberCount = i
        }
    }

    number := 0
    for primeNumberCount > 0 {
        number++
        j := 0

        for i:=1; i <= number; i++ {
            if number % i == 0 {
                j++
            }
        }

        if j == 2 {
            primeNumberCount--
        }

    }

    fmt.Printf("The latest prime number: %d\n", number)
}

File: prime-number/f#/cmd.fs
[<EntryPoint>]
let main args =
    let mutable primeNumberCount : int32 = if (args.Length > 0) then int args[0] else 100

    let mutable number : int32 = 0
    while (primeNumberCount > 0) do
        number <- number + 1
        let mutable j : int32 = 0

        for i = 1 to number do
            if (number % i = 0) then
                j <- j + 1

        if (j = 2) then
            primeNumberCount <- primeNumberCount - 1

    printfn "The latest prime number: %d\n" number
    0
